apps/backend/src/http-ping-bucket/schedule/http-ping-bucket-scheduler.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Logger } from '@logdash/js-sdk';
+import { Injectable } from '@nestjs/common';
+import { Cron, CronExpression } from '@nestjs/schedule';
+import { ClickhouseUtils } from '../../clickhouse/clickhouse.utils';
+import { HttpPingBucketAggregateService } from '../aggregate/http-ping-bucket-aggregate.service';
+import { HttpPingBucketEntity } from '../core/entities/http-ping-bucket.entity';
+import { CreateHttpPingBucketDto } from '../write/dto/create-http-ping-bucket.dto';
+import { HttpPingBucketWriteService } from '../write/http-ping-bucket-write.service';
+
+@Injectable()
+export class HttpPingBucketSchedulerService {
+  constructor(
+    private readonly httpPingBucketAggregateService: HttpPingBucketAggregateService,
+    private readonly httpPingBucketWriteService: HttpPingBucketWriteService,
+    private readonly logger: Logger,
+  ) {}
+
+  @Cron(CronExpression.EVERY_HOUR)
+  public async createBucketsForPreviousHour(): Promise<void> {
+    try {
+      await this.aggregatePingsIntoBuckets();
+    } catch (error) {
+      this.logger.error('Error creating HTTP ping buckets:', { errorMessage: error.message });
+    }
+  }
+
+  private async aggregatePingsIntoBuckets(): Promise<void> {
+    const now = new Date();
+    const previousHourStart = new Date(
+      now.getFullYear(),
+      now.getMonth(),
+      now.getDate(),
+      now.getHours() - 1,
+    );
+    const previousHourEnd = new Date(
+      now.getFullYear(),
+      now.getMonth(),
+      now.getDate(),
+      now.getHours(),
+    );
```

Comment:
I think it would be much more stable to manually compute start of current hour (XX:00:00), then subtract 1 millisecond and call it `previousHourEnd`. Then subtract 1 hour from and add 1 millisecond that and call that `previousHourStart`. This way we will always include everything. I feel there might be some extreme edge cases if we don't do that.

In mikrus CPU is sometimes 100% blocked and we get INSANE problems with performance. You yourself witnessed how the whole `cron` package behaved very strange in limited resources environments.

Reply:
Changed to
```typescript
    const now = new Date();
    now.setMinutes(0, 0, 0);
    const previousHourStart = subHours(now, 1);
    const previousHourEnd = now;
```